core/src/main/java/com/digitalpebble/stormcrawler/protocol/file/FileResponse.java

```diff
@@ -0,0 +1,176 @@
+        if (!file.canRead()) {
+            statusCode = HttpStatus.SC_UNAUTHORIZED;
+            return;
+        }
+
+        if (!file.equals(file.getCanonicalFile())) {
+            metadata.setValue(HttpHeaders.LOCATION, file.getCanonicalFile()
+                    .toURI().toURL().toString());
+            statusCode = HttpStatus.SC_MULTIPLE_CHOICES;
+            return;
+        }
+
+        if (file.isDirectory()) {
+            getDirAsHttpResponse(file);
+        } else if (file.isFile()) {
+            getFileAsHttpResponse(file);
+        } else {
+            statusCode = HttpStatus.SC_INTERNAL_SERVER_ERROR;
+            return;
+        }
+
+        if (content == null) {
+            content = new byte[0];
+        }
+    }
+
+    public ProtocolResponse toProtocolResponse() throws IOException {
+        return new ProtocolResponse(content, statusCode, metadata);
+    }
+
+    private void getFileAsHttpResponse(File file) throws IOException {
+        long size = file.length();
+
+        if (size > Integer.MAX_VALUE) {
+            statusCode = HttpStatus.SC_BAD_REQUEST;
+            return;
+        }
+
+        try {
+            content = IOUtils.toByteArray(new FileInputStream(file), size);
+
+        } catch (IOException | IllegalArgumentException e) {
+            LOG.error("Exception while fetching file response {} ",
+                    file.getPath(), e);
+            statusCode = HttpStatus.SC_METHOD_FAILURE;
+            return;
+        }
+
+        metadata.setValue(HttpHeaders.CONTENT_LENGTH, new Long(size).toString());
+        metadata.setValue(HttpHeaders.LAST_MODIFIED,
+                formatDate(file.lastModified()));
+
+        statusCode = HttpStatus.SC_OK;
+    }
+
+    private void getDirAsHttpResponse(File file) {
+        content = generateSitemap(file);
+        metadata.setValue(HttpHeaders.CONTENT_TYPE, "application/xml");
+        metadata.setValue("isSitemap", "true");
+        statusCode = HttpStatus.SC_OK;
+    }
+
+    private static String formatDate(long date) {
+
+        return dateFormat.format(new Date(date));
+    }
+
+    private byte[] generateSitemap(File dir) {
+
+        File[] files = dir.listFiles();
+        StringBuilder sb = new StringBuilder(
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<urlset xmlns=\"http://www.sitemaps.org/schemas/sitemap/0.9\">\n");
+        sb.append("<url>\n  <loc>").append("file:\\").append(dir.getPath())
+                .append("\\</loc>\n").append("  <lastmod>")
+                .append(formatDate(dir.lastModified()))
+                .append("</lastmod>\n</url>\n");
+        for (File file : files) {
+            sb.append("<url>\n  <loc>").append("file:\\")
+                    .append(file.getPath()).append("</loc>\n")
+                    .append("  <lastmod>")
+                    .append(formatDate(file.lastModified()))
+                    .append("</lastmod>\n</url>\n");
+        }
+        if (fileProtocol.getCrawlParent()) {

```

Comment:
will remove getCrawlParent() - if users want to crawl from the root then they should set it as seed